examples/neural_compressor/text-classification/intent-classification/run_clinc.py

```diff
@@ -0,0 +1,674 @@
+
+    default_config = os.path.join(
+        os.path.abspath(os.path.join(__file__, os.path.pardir, os.path.pardir, os.path.pardir)), "config"
+    )
+
+    if optim_args.apply_distillation:
+
+        class SetFitModelTraining(torch.nn.Module):
+            def __init__(self, model, tokenizer):
+                super().__init__()
+                self.model = SetFitModel(model)
+                self.tokenizer = tokenizer
+                if hasattr(model, "config"):
+                    self.config = model.config
+
+            def forward(self, sentences=None, *args, **kwargs):
+                assert (
+                    isinstance(sentences, (tuple, list)) and len(sentences) == 2
+                ), "sentences should be a tuple or a list with 2 sentences string."
+                inputs = self.tokenizer(
+                    sentences[0] + sentences[1],
+                    padding=padding,
+                    max_length=max_seq_length,
+                    truncation=True,
+                    return_tensors="pt",
+                )
+                embeddings = self.model(**inputs)
+                length = len(embeddings) // 2
+                return {"logits": torch.cosine_similarity(embeddings[:length], embeddings[length:]), "loss": 0}
+
+        if optim_args.teacher_model_name_or_path is None:
+            raise ValueError("A teacher model is needed to apply distillation.")
+
+        if not training_args.do_train:
+            raise ValueError("do_train must be set to True for distillation.")
+
+        teacher_config = AutoConfig.from_pretrained(optim_args.teacher_model_name_or_path)
+        teacher_tokenizer = AutoTokenizer.from_pretrained(
+            optim_args.teacher_model_name_or_path,
+            use_fast=model_args.use_fast_tokenizer,
+        )
+        teacher_model = AutoModel.from_pretrained(
+            optim_args.teacher_model_name_or_path,
+            from_tf=bool(".ckpt" in model_args.model_name_or_path),
+            config=teacher_config,
+        )
+        teacher_model = SetFitModelTraining(teacher_model, teacher_tokenizer)
+
+        teacher_model.to(training_args.device)
+
+        distillation_config = IncDistillationConfig.from_pretrained(
+            optim_args.distillation_config if optim_args.distillation_config is not None else default_config,
+            config_file_name="distillation.yml",
+            cache_dir=model_args.cache_dir,
+        )
+
+        examples = raw_datasets["train"]["text"]
+        examples_duplicate_ratio = 100
+        examples_duplicate = []
+        for i in range(int(examples_duplicate_ratio)):
+            examples_duplicate.extend(examples)
+        examples_duplicate.extend(
+            examples[: int(len(examples) * (examples_duplicate_ratio - int(examples_duplicate_ratio)))]
+        )
+        if data_args.max_train_samples is not None:
+            examples_duplicate = raw_datasets["train"].select(range(data_args.max_train_samples))["text"]
```

Comment:
Shouldn't this be before computing the `examples_duplicate` ? Otherwise why compute them to disregard them later ?

Reply:
Also it would be nice to have comments describing this examples duplicate generation (why it's useful)

Reply:
This way is to keep number of examples exactly the number assigned to `max_train_samples`.

I will add comments to explain the reason for duplicating the examples. Thanks for the great suggestion.

Reply:
Thanks that would be super helpful to add! Currently `examples_duplicate` is overwritten after the examples duplicate generation if `max_train_samples` is specified. This should be modified I think

Reply:
If we first select 10 samples as `max_train_samples=10`, then do 100 times duplication, there would be 1000 samples in `examples_duplicate`, I think this is somewhat misleading as actual number of samples is not equal to the set `max_train_samples`.

Reply:
Then why do the examples duplication in the case where `data_args.max_train_samples` is not `None` ? (as it will be discarded). Also it would make sense to leave the possibility for users to modify `examples_duplicate_ratio`  (and to explain it in the README)

Reply:
Sure, I'll modify the code and readme accordingly. Thanks for the wonderful advise!